docs/pages/router/create-pages.mdx

```diff
@@ -112,3 +112,20 @@ export default function Page() {
   return <Text>Blog post: {slug}</Text>;
 }
 ```
+
+## Non-route files
+
+Your non-route files such as components and hooks cannot be placed within the `app` directory. For smaller projects may introduce seemly extra folder structure, but this rule helps prevents refactors as you adopt more advanced navigation patterns.
```

Comment:
Might I suggest rewording this? This is how I usually explain it:

> Every file and folder in the `/app` folder is either a `_layout` file, or a route in your app. This means that other files like your components, hooks, utilities etc **cannot be placed in the `/app` folder** because they're neither screens nor layout files.
> 
> A typical folder structure for a project using Expo Router might look something like this:
> 
> ```sh
> /app
> /components
> /utils
> /hooks
> app.json
> package.json
> ```

Also I'd consider adding something like this in the end since it's sure to come up:

> If you absolutely have to have one top level folder for all your code (such as a `src` folder) you can do this with a [config plugin](https://github.com/expo/universe) but we **don't recommend it** and will not accept bug reports regarding projects with custom root directories.